source/protocol/lsp-types.adb

```diff
@@ -166,6 +168,32 @@ package body LSP.Types is
       return Ada.Strings.UTF_Encoding.Wide_Strings.Encode (Wide);
    end To_UTF_8_String;
 
+   -------------------------------
+   -- To_UTF_8_Unbounded_String --
+   -------------------------------
+
+   function To_UTF_8_Unbounded_String
+     (Value : LSP_String) return GNATCOLL.JSON.UTF8_Unbounded_String
+   is
+      Wide : constant Wide_String := To_Wide_String (Value);
+      Res  : Ada.Strings.Unbounded.Unbounded_String;
+
+      Chunk_Size    : constant := 512;
+      Current_Index : Natural := Wide'First;
+      Next_Index    : Natural;
+   begin
+      --  Perform the encoding chunk by chunk, so as not to blow the stack
+      loop
+         Next_Index := Natural'Min (Current_Index + Chunk_Size, Wide'Last);
+         Ada.Strings.Unbounded.Append
+           (Res, Ada.Strings.UTF_Encoding.Wide_Strings.Encode
+              (Wide (Current_Index .. Next_Index)));
```

Comment:
One more idea. We can avoid potentially large `Wide` string on the stack if replace `Wide (a .. b)` with `Slice` function from Wide_Unbounded_Strings package.

Reply:
Done!